src/nlotrajectories/core/sdf/l4casadi.py

```diff
@@ -101,7 +104,25 @@ def train(
             for xb, yb in train_loader:
                 xb, yb = xb.to(self.device), yb.to(self.device)
                 pred = self.model(xb)
-                loss = loss_fn(pred, yb)
+
+                # Compute the MSE loss
+                mse_loss = loss_fn(pred, yb)
+                # Compute the surface loss
+                if surface_loss_weight > 0:
+                    surface_loss_value = surface_loss(
+                        yb.detach().cpu().numpy(),
```

Comment:
With detach gradient can not pass through this loss
```suggestion
                        yb
```